doc/nrf/app_dev/device_guides/nrf54l/ecies_x25519.rst

```diff
@@ -0,0 +1,75 @@
+.. _ug_nrf54l_ecies_x25519:
+
+MCUboot AES Image Encryption with ECIES-X25519 Key Exchange
+###########################################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+
+MCUboot on the nRF54L15 SoC can support encrypted images using AES.
+Images are encrypted using AES, and ECIES-X25519 is used for key delivery (exchange) within the image.
+When image encryption is enabled, you can choose to upload either signed images or encrypted images to be swapped during boot.
+If MCUboot finds an encrypted image in the secondary slot, it decrypts the image during the slot swap process.
+An image that was encrypted before being swapped into the primary slot is re-encrypted if the swap is later reverted.
+
+Limitations
+***********
+
+The current implementation has the following limitations:
+
+* On the nRF54L15 SoC, ECIES-X25519 key exchange is supported only when the ED25519 signature algorithm is selected.
+  This is the default configuration.
+* Encryption is not supported when using MCUboot in Direct-XIP mode.
+* Storing the ECIES-X25519 device private key in the Key Management Unit (KMU) is not supported at this time.
+* SHA-256 is currently used by HMAC and HKDF, but this may change in future releases.
+
+HMAC/HKDF impact on TLV and Key Exchange
+****************************************
+
+An encrypted image includes a TLV that contains the public key used for ECIES-X25519 key exchange, the encrypted AES key, and the MAC tag of the encrypted key.
+The key used to encrypt the AES key is derived using HKDF, and the MAC tag is generated using HMAC.
+Both HKDF and HMAC currently use the SHA-256 hash algorithm.
+
+While the use of SHA-256 does not pose a security concern and has minimal impact on performance, it increases code size because SHA-256 support must be included alongside SHA-512, which is already used by the ED25519 signature algorithm.
+
+Pre-installed MCUboot instances will not be able to boot images that use TLVs generated with different hash  algorithms that might be supported by this implementation in the future.
```

Comment:
General note: not sure if we should declare what might be supported in the future. 
```suggestion
Additionally, pre-installed MCUboot instances will not be able to boot images that use TLVs generated with different hash algorithms.
```